File: examples/go/scan.go
// A runnable Go example: scan a small universe through the binding.
//
//	cargo build --release -p wickra-screener-c
//	# stage the library under bindings/go/lib/<goos>_<goarch>/ (CI does this)
//	cd examples/go && go run .
package main

import (
	"fmt"

	wickra "github.com/wickra-lib/wickra-screener-go"
)

const spec = `{"universe":["AAA","BBB"],"condition":{"type":"cmp",` +
	`"left":{"kind":"price","field":"close"},"op":"gt",` +
	`"right":{"kind":"const","value":10.0}}}`

const cmd = `{"cmd":"scan","data":{` +
	`"AAA":[{"time":1,"open":5,"high":5,"low":5,"close":5,"volume":1}],` +
	`"BBB":[{"time":1,"open":15,"high":15,"low":15,"close":15,"volume":1}]}}`

func main() {
	screener, err := wickra.New(spec)
	if err != nil {
		panic(err)
	}
	defer screener.Close()

	report, err := screener.Command(cmd)
	if err != nil {
		panic(err)
	}

	fmt.Println("wickra-screener", wickra.Version())
	fmt.Println(report)
}

File: examples/java/Scan.java
// A runnable Java example: scan a small universe through the binding.
//
//   cargo build -p wickra-screener-c
//   mvn -f bindings/java/pom.xml -q package -DskipTests
//   javac -cp bindings/java/target/classes examples/java/Scan.java -d examples/java/out
//   java --enable-native-access=ALL-UNNAMED \
//        -Dnative.lib.dir=target/debug \
//        -cp "bindings/java/target/classes;examples/java/out" Scan
import org.wickra.screener.Screener;

public final class Scan {
    private static final String SPEC =
            "{\"universe\":[\"AAA\",\"BBB\"],\"condition\":{\"type\":\"cmp\","
                    + "\"left\":{\"kind\":\"price\",\"field\":\"close\"},\"op\":\"gt\","
                    + "\"right\":{\"kind\":\"const\",\"value\":10.0}}}";

    private static String candle(int close) {
        return "{\"time\":1,\"open\":" + close + ",\"high\":" + close
                + ",\"low\":" + close + ",\"close\":" + close + ",\"volume\":1}";
    }

    public static void main(String[] args) {
        try (Screener screener = new Screener(SPEC)) {
            String cmd = "{\"cmd\":\"scan\",\"data\":{"
                    + "\"AAA\":[" + candle(5) + "],"
                    + "\"BBB\":[" + candle(15) + "]}}";
            String response = screener.command(cmd);
            System.out.println("wickra-screener " + Screener.version());
            System.out.println(response);
        }
    }
}
